ocaml/typing/typecore.ml

```diff
@@ -3065,10 +3065,10 @@ let rec check_counter_example_pat
       in
       map_fold_cont type_label_pat fields
         (fun fields -> mkp k (Tpat_record (fields, closed)))
-  | Tpat_array (mut, tpl) ->
-      let ty_elt = solve_Ppat_array ~refine loc env mut expected_ty in
+  | Tpat_array (mut, _, tpl) ->
+      let ty_elt, arg_sort = solve_Ppat_array ~refine loc env mut expected_ty in
```

Comment:
I don't know this function well. I think it would be safest to assert that the returned sort is the same as the original sort. That way, in case this is wrong, we'll know right away.

Reply:
Sounds good! I added an assert with call to `Jkind.Sort.equate`.